### PR TITLE
Display unparsed datafiles

### PR DESCRIPTION
**Pull request by dandavies99**
**To be merged after #208** 

This allows the Experiment detail view to be displayed even if one of the data files isn't parsed. Currently the view assumes that for each ExperimentDataFile there will be data to plot. Now it checks that there is before trying to access columns. 

This is a draft PR for now - I would like to tidy up the logic in `battDB.views.ExperimentView`. There are a lot of methods that are used to generate the plots, which could probably be simplified. 

**UPDATE:** This is now ready for review but I expect we can still simplify the views. Interested to see if you have any suggestions @AdrianDAlessandro ...

### CODE REVIEW

**Comment by AdrianDAlessandro** [comment]
Can you give me an example file that this works for?

**Comment by AdrianDAlessandro** [comment]
The "Parsed data preview" heading should either be removed or it should have some message to say there is no data to preview. It's currently added at line 140 in experiment.html

**Comment by dandavies99** [comment]
> Can you give me an example file that this works for?

It should work for any valid (according to mime type and extension) file, but just selecting no parser when uploading. 

> The "Parsed data preview" heading should either be removed or it should have some message to say there is no data to preview. It's currently added at line 140 in experiment.html

Good point. I think I'll add a message saying "This file was uploaded without parsing" or something.

File: battDB/views.py [hunk 1]
```diff
@@ -152,6 +152,7 @@ def post(self, request, *args, **kwargs):
                 else:
                     obj.status = "private"
                 self.object = form.save()
+
             # Save individual parameters from inline form
             if formset.is_valid():
```
Comment (AdrianDAlessandro) [inline battDB/views.py:157]:
```suggestion
            if formset.is_valid() and formset[0].instance.file:
```

**Comment by dandavies99** [comment]
The latest commit gives a slightly better message to the user, explaining that the field is required. This was done by updating the clean() method of the model (this is where the "file has no file.." error was coming from) and specifying a minimum number of forms (1) allowed in the formset. 

image

**Review by AdrianDAlessandro — APPROVED** [review]
Interesting! Somewhat unintuitive reason for the formset being valid, it's like the minimum number of forms allowed is the minimum number of _valid_ forms allowed. So if it includes an invalid form but the minimum isn't set then it counts as a valid formset.

**Comment by dandavies99** [comment]
Yeah it's somewhat confusing.

**Pull request merged**